pkg/mysql/mysql.go

```diff
@@ -89,20 +89,25 @@ func ConfigMapName(cr *apiv1alpha1.PerconaServerMySQL) string {
 	return Name(cr)
 }
 
+func AutoConfigMapName(cr *apiv1alpha1.PerconaServerMySQL) string {
+	return "auto-" + Name(cr)
+}
+
 func MatchLabels(cr *apiv1alpha1.PerconaServerMySQL) map[string]string {
 	return util.SSMapMerge(cr.MySQLSpec().Labels,
 		map[string]string{apiv1alpha1.ComponentLabel: componentName},
 		cr.Labels())
 }
 
-func StatefulSet(cr *apiv1alpha1.PerconaServerMySQL, initImage, configHash string) *appsv1.StatefulSet {
+func StatefulSet(cr *apiv1alpha1.PerconaServerMySQL, initImage, configHash, autoConfigHash string) *appsv1.StatefulSet {
 	labels := MatchLabels(cr)
 	spec := cr.MySQLSpec()
 	replicas := spec.Size
 	t := true
 
 	annotations := make(map[string]string)
 	annotations["percona.com/configuration-hash"] = configHash
+	annotations["percona.com/auto-configuration-hash"] = autoConfigHash
```

Comment:
why do we need it? as far as I see only thing that affects auto config is resources and if resources are updated the statefulset will be restarted anyway

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/74/commits/07fa2f2ed36f7977484ff31f7bf8869e15c2ac00